swarm/network/stream/common_test.go

```diff
@@ -134,6 +134,9 @@ func netStoreAndDeliveryWithAddr(ctx *adapters.ServiceContext, bucket *sync.Map,
 	bucket.Store(bucketKeyDB, netStore)
 	bucket.Store(bucketKeyDelivery, delivery)
 	bucket.Store(bucketKeyFileStore, fileStore)
+	// for the kademlia object, we use the global key from the simulation package,
+	// as the simulation will try to access it in the WaitTillHealthy with that key
+	bucket.Store(simulation.BucketKeyKademlia, kad)
```

Comment:
I did not realise this requirement of using simulation/waitTillHealthy. Since this resulted in a week of delay for Vlad/Louis, I think we should handle this some other way or document it better.

Reply:
> Since this resulted in a week of delay for Vlad/Louis

? @zelig

Reply:
Wow so long delay, I thought that I added some comments regarding that, but this comment is very badly written https://github.com/ethereum/go-ethereum/blob/master/swarm/network/simulation/kademlia.go#L32. I am sorry for causing problems of this kind. Maybe we should have a better approach for this functionality, it is not clear at all, causing a lot of pain. Sorry.

Reply:
This example https://godoc.org/github.com/ethereum/go-ethereum/swarm/network/simulation#example-Simulation-WaitTillHealthy may be of some help, but still BucketKeyKademlia/WaitTillHealthy usage over time shows that it is a very bad design.